### PR TITLE
Generate `dynCall` function for all signature used by `invoke` functions.

### PR DESCRIPTION
**Pull request by sbc100**
Previously we were only creating `dynCall` functions for signatures
that we have statically in the table.  However for dynamic linking we
may need to invoke functions that we don't have table entries for
(e.g. table entries from a different module).

### CODE REVIEW

File: src/wasm-emscripten.h [hunk 1]
```diff
@@ -66,9 +66,12 @@ class EmscriptenGlueGenerator {
   Global* getStackPointerGlobal();
   Expression* generateLoadStackPointer();
   Expression* generateStoreStackPointer(Expression* value);
+  void generateDynCallThunk(std::string sig);
   void generateStackSaveFunction();
   void generateStackAllocFunction();
   void generateStackRestoreFunction();
+
+  std::unordered_set<std::string> sigs;
```
Comment (kripken) [inline src/wasm-emscripten.h]: maybe add a comment here saying this is all the signatures seen so far and that we've generated dynCalls for?

File: src/wasm/wasm-emscripten.cpp [hunk 2]
```diff
@@ -334,41 +334,44 @@ inline void exportFunction(Module& wasm, Name name, bool must_export) {
   wasm.addExport(exp);
 }
 
+void EmscriptenGlueGenerator::generateDynCallThunk(std::string sig) {
+  auto* funcType = ensureFunctionType(sig, &wasm);
+  if (!sigs.insert(sig).second) {
+    return; // sig is already in the set
+  }
+  Name name = std::string("dynCall_") + sig;
+  if (wasm.getFunctionOrNull(name) || wasm.getExportOrNull(name)) {
```
Comment (kripken) [inline src/wasm/wasm-emscripten.cpp]: just checking for an export is enough, isn't it? I'd remove the function check, as it doesn't help, and can be confusing if we change things in the future.

Comment (sbc100) [reply src/wasm/wasm-emscripten.cpp]: I don't know, this change only moves this code, doesn't modify it.  So I'd rather do that as a followup if that ok?

Comment (kripken) [reply src/wasm/wasm-emscripten.cpp]: Oh ok, I didn't realize it was just moved.

**Pull request merged**